lib/bootsnap/load_path_cache/core_ext/kernel_require.rb

```diff
@@ -35,7 +35,7 @@ def require(path)
 
   alias_method(:require_relative_without_bootsnap, :require_relative)
   def require_relative(path)
-    location = caller_locations(1..1).first
+    location = caller_locations.find { |x| x.base_label != "require_relative" }
```

Comment:
Hum this increase the cost quite significantly.

Out of curiosity what else decorates `require_relative`? Maybe we could do something like `caller_locations(1..3)` to support up to two extra decorators?

Reply:
Honestly, not aware of any public Gem decorate it, but our projects internally is about to decorate it to track some state change pre/post file load, and it's blocked by the decorator in bootnap. 

Only traverse the first few caller seems fine to me. The chaining of require_relative may not happen that often probably.

Reply:
I'll have to look on Monday if we even still need to decorate `require_relative`. I think it we might be able to remove this decorator entirely.

Reply:
Ok, so this is the change I remembered that may allow us to remove the `require_relative` decorator: https://github.com/ruby/ruby/commit/79a4484a072e9769b603e7b4fbdb15b1d7eccb15. It shipped with Ruby 3.1, so that wouldn't remove it for everyone, but I'm much less concerned about a small perf impact if it doesn't apply to the newest rubies.

The decorator was added in https://github.com/Shopify/bootsnap/pull/136, I'll have to dig more to see what it would take to remove it on 3.1+.

Reply:
Hum, the more I dig into https://github.com/Shopify/bootsnap/pull/136, the more it seems like we tried to limit a Ruby bug more than a Bootsnap bug.

Assuming `a/test.rb` and `b/test.rb` symlink each others, the following load the file twice on all rubies but 3.1:

```ruby
require_relative "b/test.rb"
require_relative "a/test.rb"
```

So I don't think we should have done this at all.